docs/dev/README.md

```diff
@@ -0,0 +1,218 @@
+    *   AWS_TEST_HTTP_PROXY_BASIC_HOST - host address of the proxy to use for tests that make open connections to the proxy with basic authentication
+    *   AWS_TEST_HTTP_PROXY_BASIC_PORT - port to use for tests that make open connections to the proxy with basic authentication
+    *   AWS_TEST_BASIC_AUTH_USERNAME - username to use when using basic authentication to the proxy
+    *   AWS_TEST_BASIC_AUTH_PASSWORD - password to use when using basic authentication to the proxy
+*   S3
+    *   AWS_TEST_S3 - set to any value to enable S3 tests.
+        Unfortunately, at this time these tests can only be run by members of the
+        Common Runtime team, due to use of hardcoded paths. TODO:
+        alter tests so anyone can run them.
+
+## Code Formatting
+
+We use automatic code formatters in this project and pull requests will fail unless
+the code is formatted correctly.
+
+`autopep8` is used for python code. You should have installed via `requirements-dev.txt` above.
+
+For C code `clang-format` is used (specifically version 9).
+To install this on Mac using homebrew, run:
+```sh
+(.venv) $ brew install llvm@9
+```
+
+Use helper scripts to automatically format your code (or configure your IDE to do it):
+```sh
+# format all code
+(.venv) $ ./scripts/format-all.py
+
+# just format Python files
+(.venv) $ ./scripts/format-python.py
+
+# just format C files
+(.venv) $ ./scripts/format-c.py
+```
+
+## Using an IDE
+### Using Visual Studio Code (VSCode)
+
+1)  Install the following extensions:
+    *   Python (Microsoft)
+    *   C/C++ (Microsoft)
+    *   Code Spell Checker (Street Side Software) - optional
+
+2)  Open the `aws-crt-python/` folder.
+
+3)  Edit workspace settings: `cmd+shift+P -> Preferences: Open Workspace Settings`
+    *   Python: Default Interpreter Path - ("python.defaultInterpreterPath" in json view)
+        set the absolute path to Python in your virtual environment. For example: `/Users/janedoe/dev/aws-crt-python/.venv/bin/python`
+        *   Note that the VSCode terminal ignores this setting and will not use your virtual environment by default.
+            You must manually run `source .venv/bin/activate` each time you start using the terminal.
+            Or use the command `cmd+shift+P -> Python: Create Terminal`.
+    *   C_Cpp > Default: Include Path - ("C_Cpp.default.includePath" in json view)
+        *   Add item - set path to Python's C headers.
+            For example: `/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10`
+        *   This is optional, it helps IntelliSense when viewing C files.
+
+4)  Add helpful tasks you can run via `cmd+shift+P -> Tasks: Run Task`
+    *   Copy [this file](vscode/launch.json) to `aws-crt-python/.vscode/tasks.json` for the following tasks:
+        * install - pip install in develop mode. `cmd+shift+B` is a special shortcut for this task
+        * format-python - format all .py files
```

Comment:
Super minor nitpick: Put command names in code blocks:
* ` install` - pip install ...
* `format-python` - format all ...